operations/setup.go

```diff
@@ -16,77 +18,117 @@ const (
-				return errors.Wrap(err, "problem configuring error handler")
-			}
+		senders = append(senders, send.NewBufferedSender(sender, loggingBufferDuration, loggingBufferCount))
+	}
+
+	if conf.Slack.Options != nil {
+		if err = conf.Slack.Options.Validate(); err != nil {
+			return nil, errors.Wrap(err, "non-nil slack configuration is not valid")
+		}
+
+		if conf.Slack.Token == "" || conf.Slack.Level == "" {
+			return nil, errors.Wrap(err, "must specify slack token and threshold")
+		}
 
-			defaultSenders = append(defaultSenders, send.NewBufferedSender(sender, loggingBufferDuration, loggingBufferCount))
+		lvl := send.LevelInfo{
+			Default:   logLevel.Default,
+			Threshold: level.FromString(conf.Slack.Level),
 		}
 
-		if appConf.Slack.Options != nil {
-			sconf := appConf.Slack
-			if err = sconf.Options.Validate(); err != nil {
-				return errors.Wrap(err, "non-nil slack configuration is not valid")
-			}
-
-			if sconf.Token == "" || sconf.Level == "" {
-				return errors.Wrap(err, "must specify slack token and threshold")
-			}
-
-			lvl := send.LevelInfo{
-				Default:   logLevelInfo.Default,
-				Threshold: level.FromString(sconf.Level),
-			}
-
-			sender, err = send.NewSlackLogger(sconf.Options, sconf.Token, lvl)
-			if err != nil {
-				return errors.Wrap(err, "problem constructing slack alert logger")
-			}
-			if err = sender.SetErrorHandler(send.ErrorHandlerFromSender(fallback)); err != nil {
-				return errors.Wrap(err, "problem configuring error handler")
-			}
-
-			// TODO consider using a local queue to buffer
-			// these messages
-			defaultSenders = append(defaultSenders, send.NewBufferedSender(sender, loggingBufferDuration, loggingBufferCount))
+		sender, err = send.NewSlackLogger(conf.Slack.Options, conf.Slack.Token, lvl)
+		if err != nil {
+			return nil, errors.Wrap(err, "problem constructing slack alert logger")
 		}
+		if err = sender.SetErrorHandler(send.ErrorHandlerFromSender(fallback)); err != nil {
+			return nil, errors.Wrap(err, "problem configuring error handler")
+		}
+
+		// TODO consider using a local queue to buffer
+		// these messages
+		senders = append(senders, send.NewBufferedSender(sender, loggingBufferDuration, loggingBufferCount))
 	}
 
-	return errors.WithStack(grip.SetSender(send.NewConfiguredMultiSender(defaultSenders...)))
+	return send.NewConfiguredMultiSender(senders...), nil
+}
+
+func (c *serviceConf) setup(ctx context.Context, env cedar.Environment) error {
+	err := env.Configure(c.export())
+	if err != nil {
+		return errors.Wrap(err, "problem setting up configuration")
+	}
+
+	conf := &model.CedarConfig{}
+	conf.Setup(env)
+	grip.Warning(conf.Find())
+
+	sender, err := c.getSenders(conf)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if err = grip.SetSender(sender); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func newServiceConf(numWorkers int, localQueue bool, mongodbURI, bucket, dbName string) *serviceConf {
```

Comment:
this function doesn't seem necessary since it is private

Reply:
just to ease conversion from the old function. I agree and think we can refactor it away, but I didn't want to change that just yet (the tests make things a bit complicated.)

Reply:
sounds good